src/components/SelectTokenTextField/index.tsx

```diff
@@ -72,6 +84,17 @@ export const SelectTokenTextField: React.FC<SelectTokenTextFieldProps> = ({
 
                 <Icon css={styles.getArrowIcon({ isTokenListShown })} name="arrowUp" />
               </PrimaryButton>
+              {rightMaxButton ? (
+                <TertiaryButton
+                  onClick={() => setMaxValue(rightMaxButton.valueOnClick)}
+                  small
+                  disabled={disabled}
+                  css={styles.maxButton}
+                  data-testid={!!testId && getTokenMaxButton({ parentTestId: testId })}
+                >
+                  {rightMaxButton.label}
+                </TertiaryButton>
+              ) : undefined}
```

Comment:
You don't need the ternary operator: 
```suggestion
              {rightMaxButton && (
                <TertiaryButton
                  onClick={() => setMaxValue(rightMaxButton.valueOnClick)}
                  small
                  disabled={disabled}
                  css={styles.maxButton}
                  data-testid={!!testId && getTokenMaxButton({ parentTestId: testId })}
                >
                  {rightMaxButton.label}
                </TertiaryButton>
              )}
```